x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/cells/index.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { ALERT_STATUS, ALERT_CASE_IDS } from '@kbn/rule-data-utils';
+import { CellComponentProps, SystemCellComponentMap, SystemCellId } from '../types';
+import { DefaultCell } from './default_cell';
+import { AlertLifecycleStatusCell } from './alert_lifecycle_status_cell';
+import { CasesCell } from '../cases/cell';
+
+export const systemCells: SystemCellId[] = [ALERT_STATUS, ALERT_CASE_IDS];
+
+const SystemCellFactoryComponent: React.FC<CellComponentProps> = (props) => {
```

Comment:
I created a factory for our internal cells. The factory will get the `columnId` and all required props and render the corresponding cell component.